internal/routing/ip.go

```diff
@@ -27,9 +29,15 @@ func (r *Routing) assignedIP(interfaceName string) (ip net.IP, err error) {
 	for _, address := range addresses {
 		switch value := address.(type) {
 		case *net.IPAddr:
-			return value.IP, nil
+			if (family == netlink.FAMILY_V6 && value.IP.To4() == nil) ||
+				(family == netlink.FAMILY_V4 && value.IP.To4() != nil) {
```

Comment:
Perhaps have this as a tiny function i.e. `ipMatchesFamily(ip net.IP, family int)` since it's used twice.